packages/material-ui-lab/src/TimelineDot/TimelineDot.js

```diff
@@ -1,67 +1,106 @@
+    styleProps.variant === 'filled' && {
+      borderColor: 'transparent',
+      color: theme.palette.grey[50],
+      backgroundColor: theme.palette.grey[400],
+    }),
   /* Styles applied to the root element if `color="grey"` and `variant="outlined"`. */
-  outlinedGrey: {
-    boxShadow: 'none',
-    color: theme.palette.grey.contrastText,
-    borderColor: theme.palette.grey[400],
-    backgroundColor: 'transparent',
-  },
+  ...(styleProps.color === 'grey' &&
+    styleProps.variant === 'outlined' && {
+      boxShadow: 'none',
+      color: theme.palette.grey.contrastText,
+      borderColor: theme.palette.grey[400],
+      backgroundColor: 'transparent',
+    }),
   /* Styles applied to the root element if `color="primary"` and `variant="filled"`. */
-  filledPrimary: {
-    borderColor: 'transparent',
-    color: theme.palette.primary.contrastText,
-    backgroundColor: theme.palette.primary.main,
-  },
+  ...(styleProps.color === 'primary' &&
+    styleProps.variant === 'filled' && {
+      borderColor: 'transparent',
+      color: theme.palette.primary.contrastText,
+      backgroundColor: theme.palette.primary.main,
+    }),
   /* Styles applied to the root element if `color="primary"` and `variant="outlined"`. */
-  outlinedPrimary: {
-    boxShadow: 'none',
-    backgroundColor: 'transparent',
-    borderColor: theme.palette.primary.main,
-  },
+  ...(styleProps.color === 'primary' &&
+    styleProps.variant === 'outlined' && {
+      boxShadow: 'none',
+      backgroundColor: 'transparent',
+      borderColor: theme.palette.primary.main,
+    }),
   /* Styles applied to the root element if `color="secondary"` and `variant="filled"`. */
-  filledSecondary: {
-    borderColor: 'transparent',
-    color: theme.palette.secondary.contrastText,
-    backgroundColor: theme.palette.secondary.main,
-  },
+  ...(styleProps.color === 'secondary' &&
+    styleProps.variant === 'filled' && {
+      borderColor: 'transparent',
+      color: theme.palette.secondary.contrastText,
+      backgroundColor: theme.palette.secondary.main,
+    }),
   /* Styles applied to the root element if `color="secondary"` and `variant="outlined"`. */
-  outlinedSecondary: {
-    boxShadow: 'none',
-    backgroundColor: 'transparent',
-    borderColor: theme.palette.secondary.main,
-  },
-});
+  ...(styleProps.color === 'secondary' &&
+    styleProps.variant === 'outlined' && {
+      boxShadow: 'none',
+      backgroundColor: 'transparent',
+      borderColor: theme.palette.secondary.main,
+    }),
+}));
 
-const TimelineDot = React.forwardRef(function TimelineDot(props, ref) {
-  const { classes, className, color = 'grey', variant = 'filled', ...other } = props;
+const TimelineDot = React.forwardRef(function TimelineDot(inProps, ref) {
+  const props = useThemeProps({ props: inProps, name: 'MuiTimelineDot' });
+  const { className, color = 'grey', variant = 'filled', ...other } = props;
 
   const themeVariantsClasses = useThemeVariants(
```

Comment:
No need for `themeVariantsClasses` anymore, they are already handled by the new `styled()` utility.